src/tests/test_kube_api.py

```diff
@@ -101,76 +423,29 @@ def unbound_highly_available_infraenv(
-        )
+        yield self.kube_api_test_prepare_late_binding_cluster(kube_api_context.api_client, cluster_config, 1)
 
     @pytest.fixture
     @JunitFixtureTestCase()
     def unbound_highly_available_cluster(self, kube_test_configs_highly_available, kube_api_context):
         cluster_config, _ = kube_test_configs_highly_available
-        yield kube_api_test_prepare_late_binding_cluster(
-            kube_api_context=kube_api_context, cluster_config=cluster_config, num_controlplane_agents=3
-        )
+        yield self.kube_api_test_prepare_late_binding_cluster(kube_api_context.api_client, cluster_config, 3)
 
     @pytest.fixture
     @JunitFixtureTestCase()
     def unbound_single_node_cluster_hold_installation(self, kube_test_configs_single_node, kube_api_context):
         cluster_config, _ = kube_test_configs_single_node
-        yield kube_api_test_prepare_late_binding_cluster(
-            kube_api_context=kube_api_context,
-            cluster_config=cluster_config,
-            num_controlplane_agents=1,
-            hold_installation=True,
-        )
-
-    @JunitTestSuite()
-    @pytest.mark.kube_api
-    def test_kube_api_ipv4(self, kube_test_configs_single_node, kube_api_context, get_nodes):
-        cluster_config, tf_config = kube_test_configs_single_node
-        kube_api_test(kube_api_context, get_nodes(tf_config, cluster_config), cluster_config)
-
-    @JunitTestSuite()
-    @pytest.mark.kube_api
-    def test_kube_api_ipv6(self, kube_test_configs_single_node, kube_api_context, proxy_server, get_nodes):
-        cluster_config, tf_config = kube_test_configs_single_node
-        tf_config.is_ipv6 = True
-        tf_config.is_ipv4 = False
-
-        kube_api_test(
-            kube_api_context, get_nodes(tf_config, cluster_config), cluster_config, proxy_server, is_ipv4=False
-        )
-
-    @JunitTestSuite()
-    @pytest.mark.kube_api
-    def test_capi_provider(self, kube_test_configs_highly_available, kube_api_context, get_nodes):
-        cluster_config, tf_config = kube_test_configs_highly_available
-        capi_test(kube_api_context, get_nodes(tf_config, cluster_config), cluster_config)
-
-    @staticmethod
-    def _bind_all(cluster_deployment, agents):
-        for agent in agents:
-            agent.bind(cluster_deployment)
-
-    @staticmethod
-    def _wait_for_install(agent_cluster_install, agents):
-        agent_cluster_install.wait_to_be_ready(True)
-        agent_cluster_install.wait_to_be_installing()
-        Agent.wait_for_agents_to_install(agents)
-        agent_cluster_install.wait_to_be_installed()
-
-    @staticmethod
-    def _set_agent_cluster_install_machine_cidr(agent_cluster_install, nodes):
-        machine_cidr = nodes.controller.get_primary_machine_cidr()
-        agent_cluster_install.set_machinenetwork(machine_cidr)
+        api_client = kube_api_context.api_client
+        yield self.kube_api_test_prepare_late_binding_cluster(api_client, cluster_config, 1, hold_installation=True)
```

Comment:
also on those kind of places. ``hold_installation=True`` is great, ``1`` is just not understandable